src/btree/bt_curnext.c

```diff
@@ -439,8 +439,14 @@ __cursor_row_next(WT_CURSOR_BTREE *cbt, bool newpage, bool restart, size_t *skip
              * If an upper bound has been set ensure that the key is within the range, otherwise
              * early exit.
              */
-            if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER))
-                WT_RET(__wt_btcur_bounds_early_exit(session, cbt, true, key_out_of_bounds));
+            if (F_ISSET(cursor, WT_CURSTD_BOUND_UPPER)) {
```

Comment:
As we head towards deprecating prefix search near can we use `cbt->iface` here so its consistent with prev? Same for the check below.